app/models/repo/challenge.rb

```diff
@@ -29,5 +29,9 @@ class Challenge < ApplicationRecord
     validates :registration_at, comparison: { less_than: :start_at }, allow_nil: true, if: :start_at
     validates :start_at, comparison: { less_than: :finish_at }, allow_nil: true, if: :finish_at
     validates :finish_at, comparison: { greater_than: :start_at }, allow_nil: true, if: :start_at
+
+    def opened?
+      (registration_at.to_datetime..finish_at.to_datetime).cover?(Time.zone.now)
```

Comment:
no business login in models, please. Use policy facade

Reply:
Thanks for this point. Firstly, I put it into ChallengePolicy, but then I found that it would be good to use it in [view](https://github.com/DevChallengeIT/devchallenge-platform/pull/41/files#diff-6f2957f2168b3bb52994a7a828997d3a1cdcbb0afc2d09d04018786fe7201cbcR4) to check if a challenge is opened.

So I can move this method to challenge_policy and make it public (ex. challenge_opened?), delegate it to Competition, then call it from the view. But I guess it will break the srp of challenge_policy which is about relation user-challenge. We can create a facade related only to the challenge inside logic/competition and use it in ChallengePolicy and in view. What do you think?

Reply:
We do not need a new state "opened". 
Already exists can_join? policy. Just add time-based rules there.

Also, think about:
- you need to show the message "registration is not opened yet"
- "registration already closed"

Probably we need to extend the policy and create a UI component with reason translation.

Reply:
Please look if I correctly understand your idea.